src/utility/png.cpp

```diff
@@ -95,9 +94,14 @@ bool png::write_png(std::istream& in, uint32_t size, uint32_t dots_per_inch,
         static constexpr int32_t bits_per_byte = 8;
         static constexpr uint8_t margin_value = 0xff;
 
-        // TODO: unguarded overflow conditions.
         const auto margin_size = margin * size;
+        if ((size == 0) || (margin_size / size) != margin)
```

Comment:
I'm not confident these round trip checks are sufficient guards. The guard should require the domain limit.